lib/ddtrace/logging/limiter.rb

```diff
@@ -0,0 +1,54 @@
+require 'time'
+
+module Datadog
+  module Logging
+    # Limiter is a helper class for managing log message rate limits
+    class Limiter
+      def initialize
+        reset!
+      end
+
+      def rate_limit!(key, timestamp = nil, &block)
+        # If no logging rate is enabled, then immediately call the provided block, they are not rate limited
+        unless Datadog.configuration.logging.rate > 0
```

Comment:
Should this reference the global setting directly? My thinking is maybe this rate should be an attribute of the rate limiter instead. To set this value, you could either do `Limiter.new(rate)` or `limiter.rate = rate`. Then maybe global setting should just passthrough and attempt to configure the global tracer's logger accordingly.